src/testing/tree_construction.rs

```diff
@@ -50,188 +32,80 @@ pub struct Test {
     pub line: usize,
     /// The specification of the test provided in the test file
     pub spec: TestSpec,
-    /// The document tree that is expected to be parsed
+    /// The document tree as found in the spec converted to an array
     pub document: Vec<String>,
```

Comment:
Side note (unrelated to this PR) — I feel like we're doing a lot of processing of the `TestSpec`, e.g., by splitting the document into lines, and doing sorting and other processing of the lines.  I suspect a more rigorous approach would be to just take the document as it is provided in the third-party test and attempt to reproduce it from our own parsing of the input.

(That observation is not meant for this PR.)

Reply:
I think document here isn't something that would need to be in the test-spec, but is something that could be generated whenever we need to. The thing is that I'm testing the results better by generating a tree structure from our document, and compare that to the tree from the `#document` found in the test spec. One of the benefits is that the tests itself are easier because it's now nothing more than a line-by-line compare. The previous way was too complex and resulted in some false positives that we didn't catch until i've actually refactored this.